src/Microsoft.Extensions.Configuration.Json/JsonConfigurationFileParser.cs

```diff
@@ -87,15 +88,15 @@ private void VisitArray(JArray array)
             }
         }
 
-        private void VisitPrimitive(JToken data)
+        private void VisitPrimitive(JValue data)
         {
             var key = _currentPath;
 
             if (_data.ContainsKey(key))
             {
                 throw new FormatException(Resources.FormatError_KeyIsDuplicated(key));
             }
-            _data[key] = data.ToString();
+            _data[key] = data.ToString(CultureInfo.InvariantCulture);
```

Comment:
I haven't considered that. If I'm reading the source of JSON.NET right:

* `data.ToString(CultureInfo.InvariantCulture)` calls `JValue.ToString(IFormatProvider)`
* `JValue.ToString(IFormatProvider)` calls `JValue.ToString(string, IFormatProvider)`
* `JValue.ToString(string, IFormatProvider)` calls `IFormattable.ToString(string, IFormatProvider)` on the value, if the value is `IFormattable` and `object.ToString()` otherwise

On the other hand:

* `data.Value<string>()` calls `Extensions.Value<U>(IEnumerable<JToken>)`
* `Extensions.Value<U>(IEnumerable<JToken>)` calls `Extensions.Value<T, U>(IEnumerable<T>)`
* `Extensions.Value<T, U>(IEnumerable<T>)` calls `Extensions.Convert<T, U>(T)`
* `Extensions.Convert<T, U>(T)` calls `Convert.ChangeType(Object, Type, IFormatProvider)` (passing `InvariantCulture` as the `IFormatProvider`) if the value is not already the desired type
* `Convert.ChangeType(Object, Type, IFormatProvider)` calls `IConvertible.ToString(IFormatProvider)` on the value

From that, I would expect that `.Value<string>()` is probably not going to be any faster.

I don't think there are going to be any functional differences.

Considering that `.ToString(CultureInfo.InvariantCulture)` is also very clear about what it does, it seems like the better choice to me.

Is performance here important enough to actually measure it?